c/tskit/trees.c

```diff
@@ -3289,7 +3359,42 @@ tsk_treeseq_two_branch_count_stat(const tsk_treeseq_t *self, tsk_size_t state_di
     tsk_safe_free(col_repeats);
     iter_state_free(&l_state);
     iter_state_free(&r_state);
-    tsk_bit_array_free(&node_samples);
+    tsk_bitset_free(&node_samples);
+    tsk_bitset_free(&sample_sets_bits);
+    return ret;
+}
+
+static int
+check_sample_set_dups(tsk_size_t num_sample_sets, const tsk_size_t *sample_set_sizes,
+    const tsk_id_t *sample_sets, const tsk_id_t *restrict sample_index_map,
+    tsk_size_t num_samples)
+{
+    int ret;
+    tsk_size_t j, k, l;
+    tsk_id_t u, sample_index;
+    tsk_bitset_t tmp;
+
+    tsk_memset(&tmp, 0, sizeof(tmp));
+    ret = tsk_bitset_init(&tmp, num_samples, 1);
+    if (ret != 0) {
```

Comment:
Use ``goto out`` for consistency (in case we ever add something above it which needs cleanup)